src/tests/api/auth/sign-in/service.spec.ts

```
@@ -0,0 +1,108 @@
+	const validPassword = "JohnDoe@123";
+	const invalidPassword = "John@123Doe";
+
+	describe("Success", () => {
+		it("should sign-in", async () => {
+			let result: any;
+
+			refreshTokenMock.create.mockResolvedValue(refreshTokenMock.doc());
+
+			const fakeUserId = "7ee0fb6a-ce98-4499-b4f5-5738eb6c4efd";
+			userMock.getByEmail.mockResolvedValue(
+				userMock.doc({ userId: fakeUserId }),
+			);
+
+			try {
+				result = await service(
+					{
+						createRefreshTokenRepository: refreshTokenMock.create,
+						getByEmailUserRepository: userMock.getByEmail,
+					},
+					{
+						email: validEmail,
+						password: validPassword,
+					},
+				);
+			} catch (err) {
+				result = err;
+			}
+
+			expect(result).toHaveProperty("statusCode");
+			expect(result).toHaveProperty("body");
+			expect(result).toMatchObject({ statusCode: StatusCodeEnum.SUCCESS });
+			expect(typeof result.body).toBe("string");
+			expect(JSON.parse(result.body)).toMatchObject({
+				accessToken: tokenGenerator({ userId: fakeUserId }).accessToken,
+			});
+			expect(userMock.getByEmail).toBeCalled();
+			expect(refreshTokenMock.create).toBeCalled();
+		});
+	});
+
+	describe("Fail", () => {
+		it("invalid user", async () => {
+			let result: any;
+
+			refreshTokenMock.create.mockResolvedValue(refreshTokenMock.doc());
+			userMock.getByEmail.mockResolvedValue(undefined);
+
+			try {
+				result = await service(
+					{
+						createRefreshTokenRepository: refreshTokenMock.create,
+						getByEmailUserRepository: userMock.getByEmail,
+					},
+					{
+						email: validEmail,
+						password: invalidPassword,
+					},
+				);
+			} catch (err) {
+				result = err;
+			}
+
+			expect(result).toBeInstanceOf(CustomError);
+			expect(result).toHaveProperty("statusCode");
+			expect(result).toHaveProperty("body");
+			expect(userMock.getByEmail).toBeCalled();
+			expect(refreshTokenMock.create).not.toBeCalled();
+		});
+
+		it("invalid password", async () => {
```

`it("should throw an error with invalid password",`